### PR TITLE
feat(log): optimize open db log

### PR DESCRIPTION
**Pull request by lurais**
**What does this PR do?**
Add suggestion when open db fail.

**Why are these changes required?**
To improve user-friendliness.

### CODE REVIEW

File: chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java [hunk 1]
```diff
@@ -125,7 +125,8 @@ public void initDB() {
         openDatabase(options);
         alive = true;
       } catch (IOException ioe) {
-        throw new RuntimeException(String.format("Can't initialize database, %s", dataBaseName),
+        throw new RuntimeException(String.format("Can't initialize database: %s, please delete " +
+            "current database and download the database and restart the node.", dataBaseName),
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:129]: only need to download the corrupted database,for example account?

File: chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java [hunk 2]
```diff
@@ -125,7 +125,8 @@ public void initDB() {
         openDatabase(options);
         alive = true;
       } catch (IOException ioe) {
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:127]: `IOException ioe ` in there : Will there be any other types of errors here, all about database corruption?

Comment (halibobo1205) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:127]: According to your analysis, a difference should be made.

Comment (lurais) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:127]: > InvalidPathException
So what difference should be made? When a NativeDB.DBException means database corruption and an  IOException should be thrown? @halibobo1205

Comment (halibobo1205) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:127]: Sorry, I guess I wasn't clear enough, you may get ideas here #5223

Comment (lurais) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:127]: > #5223

That seems to be more precise, Just use the requirement "if (e.getMessage().contains("Corruption:")) {" should find the corruption certainly.

File: chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java [hunk 3]
```diff
@@ -142,11 +142,22 @@ private void openDatabase(Options dbOptions) throws IOException {
...
+      database = factory.open(dbPath.toFile(), dbOptions);
+      if (!this.getDBName().startsWith("checkpoint")) {
+        logger
+            .info("DB {} open success with writeBufferSize {} M, cacheSize {} M, maxOpenFiles {}.",
+                this.getDBName(), dbOptions.writeBufferSize() / 1024 / 1024,
+                dbOptions.cacheSize() / 1024 / 1024, dbOptions.maxOpenFiles());
+      }
+    } catch (IOException e) {
+      if (e.getMessage().contains("Corruption:")) {
+        throw new RuntimeException(String.format("Can't initialize database: %s, please delete " +
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:155]: Please do not throw an exception here. If you do not know what to do, please refer https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572. And, please keep the changes to the RocksDB.

Comment (lurais) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:155]: > Please do not throw an exception here. If you do not know what to do, please refer https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572. And, please keep the changes to the RocksDB.

Just exit when open db fail is ok.@halibobo1205

File: chainbase/src/main/java/org/tron/common/storage/rocksdb/RocksDbDataSourceImpl.java [hunk 4]
```diff
@@ -265,8 +265,10 @@ protected void log(InfoLogLevel infoLogLevel, String logMsg) {
           try {
             database = RocksDB.open(options, dbPath.toString());
           } catch (RocksDBException e) {
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/rocksdb/RocksDbDataSourceImpl.java:267]: Are all the errors here database corruption errors?

File: chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java [hunk 5]
```diff
@@ -142,11 +142,23 @@ private void openDatabase(Options dbOptions) throws IOException {
...
+      database = factory.open(dbPath.toFile(), dbOptions);
+      if (!this.getDBName().startsWith("checkpoint")) {
+        logger
+            .info("DB {} open success with writeBufferSize {} M, cacheSize {} M, maxOpenFiles {}.",
+                this.getDBName(), dbOptions.writeBufferSize() / 1024 / 1024,
+                dbOptions.cacheSize() / 1024 / 1024, dbOptions.maxOpenFiles());
+      }
+    } catch (IOException e) {
+      if (e.getMessage().contains("Corruption:")) {
+        logger.error("Database {} corrupted, please delete database directory({}) " +
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:155]: please refer https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572

Comment (lurais) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:155]: > please refer https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572

@halibobo1205 Just print the same message as L572? Or just exit when any exception is caught?

Comment (halibobo1205) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:155]: 1. Exit when any db exception is caught
2. Give a tip just like  [L572](https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572) when db is corrupted.

**Review by tomatoishealthy — APPROVED** [review]
LGTM, fix the CI check

File: framework/src/test/java/org/tron/common/storage/leveldb/LevelDbDataSourceImplTest.java [hunk 6]
```diff
@@ -107,6 +109,13 @@ public void testReset() {
         Args.getInstance().getOutputDirectory(), "test_reset");
     dataSource.resetDb();
     assertEquals(0, dataSource.allKeys().size());
+    assertEquals("LEVELDB", dataSource.getEngine());
```
Comment (halibobo1205) [inline framework/src/test/java/org/tron/common/storage/leveldb/LevelDbDataSourceImplTest.java:112]: Please add a test to cover this PR, do not add tests that is not relevant in this PR

File: framework/src/test/java/org/tron/common/storage/leveldb/RocksDbDataSourceImplTest.java [hunk 7]
```diff
@@ -97,6 +99,13 @@ public void testReset() {
         Args.getInstance().getOutputDirectory(), "test_reset");
     dataSource.resetDb();
     assertEquals(0, dataSource.allKeys().size());
+    assertEquals("ROCKSDB", dataSource.getEngine());
```
Comment (halibobo1205) [inline framework/src/test/java/org/tron/common/storage/leveldb/RocksDbDataSourceImplTest.java:102]: Comments as above

**Review by halibobo1205 — APPROVED** [review]

**Pull request merged**